services/QuillLMS/app/workers/copy_concept_results_to_responses_worker.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class CopyConceptResultsToResponsesWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: SidekiqQueue::MIGRATION
+
+  BATCH_SIZE=100000
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def perform(start, finish)
+    directions_cache = {}
+    instructions_cache = {}
+    previous_feedbacks_cache = {}
+    prompts_cache = {}
+    question_types_cache = {}
+
+    Response.bulk_insert(ignore: true) do |worker|
+      ConceptResult.where(id: start..finish).each do |concept_result|
+
+        directions = concept_result.metadata['directions']
+        instructions = concept_result.metadata['instructions']
+        previous_feedback = concept_result.metadata['lastFeedback']
+        prompt = concept_result.metadata['prompt']
+        question_type = concept_result.question_type
+
+        directions_cache[directions] ||= ResponseDirections.find_or_create_by!(text: directions)&.id if directions
+        instructions_cache[instructions] ||= ResponseInstructions.find_or_create_by!(text: instructions)&.id if instructions
+        previous_feedbacks_cache[previous_feedback] ||= ResponsePreviousFeedback.find_or_create_by!(text: previous_feedback)&.id if previous_feedback
+        prompts_cache[prompt] ||= ResponsePrompt.find_or_create_by!(text: prompt)&.id if prompt
+        question_types_cache[question_type] ||= ResponseQuestionType.find_or_create_by!(text: question_type)&.id if question_type
+
+        extra_metadata = Response.parse_extra_metadata(concept_result.metadata)
+
+        worker.add({
```

Comment:
Let's name this something other than `worker`, since it's slightly confusing within a file names `_worker`. Maybe `bulk_insert`?